machine/subcommands/create.py

```diff
@@ -43,6 +43,10 @@ def command(context, name, tag, type, region, machine_size, image, wait_for_ip,
     if update_dns and not config.dns_zone:
         fatal_error("Error: DNS update requested but no zone configured")
 
+    fqdn = name
+    if config.dns_zone:
```

Comment:
What happens if `dns_zone` is not configured?

Reply:
The code explodes creating the command context in main.py, before you get here.  So, I could take out the check here since it is effectively required.  I hadn't tried it at the time I wrote this and thought it was optional.

Reply:
I think it is optional. Can you make it pass empty values for the `MACHINE_FQDN` variable in that case?
Then scripts can be made to fail gracefully if they need the full host name.

Reply:
Fixed